src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Windows.cs

```diff
@@ -108,14 +108,16 @@ internal static SafeSocketHandle CreateWSASocket(AddressFamily addressFamily, So
             return CreateSocket(InnerSafeCloseSocket.Accept(socketHandle, socketAddress, ref socketAddressSize));
         }
 
-        private void InnerReleaseHandle()
+        private bool InnerReleaseHandle()
         {
             // Keep m_IocpBoundHandle around after disposing it to allow freeing NativeOverlapped.
             // ThreadPoolBoundHandle allows FreeNativeOverlapped even after it has been disposed.
             if (_iocpBoundHandle != null)
             {
                 _iocpBoundHandle.Dispose();
             }
+
+            return false;
```

Comment:
What does `return false` mean here?  Generally the result of a "ReleaseHandle" call is true for success and false for catastrophic failure.

Reply:
The return value indicates whether the connection should do an abortive close. I'm going to rename this method to `DoReleaseHandle` and add some comments.